### PR TITLE
Adding listing to Coretime Chain, Broker pallet

### PR DESCRIPTION
**Pull request by poppyseedDev**
These PR changes add a listing feature to the Broker pallet allowing the purchased cores to be listed and re-purchased.

### CODE REVIEW

File: substrate/frame/broker/src/lib.rs [hunk 1]
```diff
@@ -410,6 +420,24 @@ pub mod pallet {
...
 			core: CoreIndex,
 		},
+		/// A new listing has been created.
+		ListingCreated {
+			region_id: RegionId,
+			owner: T::AccountId,
+			price: BalanceOf<T>,
+		},
+		/// A region has been successfully sold.
+		RegionSold {
```
Comment (ggwpez) [inline substrate/frame/broker/src/lib.rs:430]: A secondary market can already be built by trading CoreTime NFTs, or? Could be traded on KodaDot for example?  
Why do we need additional functionality in here?

Comment (poppyseedDev) [reply substrate/frame/broker/src/lib.rs:430]: For the sake of simplicity.

**Review by Szegoo — CHANGES_REQUESTED** [review]
I don't believe the market should be embedded within a system chain. What you proposed here could make sense if we wanted parachains to rely on a set of individuals for Coretime procurement. 

However, if we want parachains to be truly decentralized systems, that cannot be the case. So another alternative would be for the parachain to collectively(via its governance) purchase Coretime. However, that isn't ideal either. Given that parachains are fully transparent systems, this could make front-running attacks very common.

I have written a potential solution to this problem [here](https://forum.polkadot.network/t/building-a-coretime-marketplace/6661).

**Comment by bkchr** [comment]
Yeah given the pallets and the work already being done by @Szegoo I don't see that this pull request is required. If there is functionality missing when it comes to transferring the regions etc, we should invest time on this.

Nevertheless, thank you for the pr!

**Pull request closed without merging**